holoviews/plotting/renderer.py

```diff
@@ -175,6 +187,21 @@ def html(self, obj, fmt=None, css={}):
         return tag.format(src=src, mime_type=mime_type, css=css)
 
 
+    def get_widget(self, plot, widget_type):
+        isuniform = plot.uniform
```

Comment:
Does (or should?) widgets have an 'auto' option? I think it is a little weird you can specify a ``widget_type`` just to override it. If you have 'auto', you can turn that into a concrete widget but if you specify a widget and it doesn't make sense, shouldn't you raise an exception instead of overriding it?